sympy/sets/fancysets.py

```diff
@@ -520,20 +520,20 @@ def __new__(cls, *args):
             raise ValueError("step cannot be 0")
 
         start, stop, step = slc.start or 0, slc.stop, slc.step or 1
-        try:
-            start, stop, step = [
-                w if w in [S.NegativeInfinity, S.Infinity]
-                else sympify(as_int(w))
-                for w in (start, stop, step)]
-        except ValueError:
-            raise ValueError(filldedent('''
-    Finite arguments to Range must be integers; `imageset` can define
-    other cases, e.g. use `imageset(i, i/10, Range(3))` to give
-    [0, 1/10, 1/5].'''))
+        params = []
+        for w in (start, stop, step):
+            if w in [S.NegativeInfinity, S.Infinity] or sympify(w).is_integer == True:
+                params.append(sympify(w))
+            else:
+                raise ValueError(filldedent('''
+        Finite arguments to Range must be integer symbols; `imageset` can define
+        other cases, e.g. use `imageset(i, i/10, Range(3))` to give
+        [0, 1/10, 1/5].'''))
+
+        start, stop, step = params
 
-        if not step.is_Integer:
-            raise ValueError(filldedent('''
-    Ranges must have a literal integer step.'''))
+        if not step.is_finite == True:
+            raise ValueError("step must be a finite integer symbol.")
```

Comment:
If you don't insist on the sign, too, being positive or negative, how will you determine which is the inf/sup?

Reply:
So, you mean that, if step is a symbol then how will we determine the inf/sup, well it's a valid point. So, how to do it? If `is_positive` is either `True` or `False` then it's easy. But if, `is_positive` is `None` then its hard. But if we return `Piecewise` something like, `Piecewise((step > 0, <the-inf/sup-for-positive-case>), (step < 0, <the-inf/sup-for-negative-case>), (Eq(step, 0), <the-inf/sup-for-zero-step>))`. What do you say?

Reply:
Please refer https://github.com/sympy/sympy/pull/17146#discussion_r300364521 too.

Reply:
Maybe you can use Min/Max for this:
```
Range(a, b, c).inf -> Min(a, b-c)
```